lib/srv/discovery/discovery.go

```diff
@@ -467,6 +475,10 @@ func New(ctx context.Context, cfg *Config) (*Server, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	if err := s.initTAGAzureWatchers(s.ctx, cfg); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
```

Comment:
From @marcoandredinis in https://github.com/gravitational/teleport/pull/48843#discussion_r1888362773
> Can we either move the TAGAWSWatcher here or move the `initTagAzure` inside the `initAccessGraphWatchers`?